[spa] Dialect selector broken again

The Spanish dialect selector test is once again broken. 

```
FAILED tests/test_wikipron/test_config.py::test_spanish_dialect_selection - AssertionError: assert 1 == 0
```

[Sample run here](https://app.circleci.com/pipelines/github/CUNY-CL/wikipron/1530/workflows/279703ff-9490-4cc8-ac64-3e9b1ab8d8b8/jobs/8314).

I suspect there's been yet another backend change to break this.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,10 @@ Unreleased
     removing ɾ, tʰ, y where needed. (\#555)
 -   Rescrapes General American and UK Received Pronunciation data. (\#555)
 
+### Under `src/` and elsewhere
+
+-   Updated dialect selector to use contains logic rather than exact match (\#557)
+
 [1.3.3] - 2024-07-27
 --------------------
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -81,8 +81,8 @@ branch by the author.
 If you would like to help avoid wasting free Internet resources
 (every push triggers a new CI autobuild),
 you can run the following checks locally before pushing commits:
-* `mypy --ignore-missing-imports wikipron/ tests/ data/`
-* `flake8 wikipron/ tests/`
-* `black --line-length=79 --check wikipron/ tests/ data/`
+* `mypy --ignore-missing-imports src/wikipron/ tests/ data/`
+* `flake8 src/wikipron/ tests/`
+* `black --line-length=79 --check src/wikipron/ tests/ data/`
     * You can fix any errors by running the same command without `--check`
 * `pytest tests/`

--- a/src/wikipron/config.py
+++ b/src/wikipron/config.py
@@ -175,7 +175,7 @@ class Config:
         else:
             dialect_selector = _DIALECT_XPATH_SELECTOR_TEMPLATE.format(
                 dialects_text=" or ".join(
-                    f'text() = "{d.strip()}"' for d in dialect.split("|")
+                    f'contains(text(), "{d.strip()}")' for d in dialect.split("|")
                 )
             )
             logging.info("Dialect(s): %r", dialect)

--- a/src/wikipron/config.py
+++ b/src/wikipron/config.py
@@ -175,7 +175,8 @@ class Config:
         else:
             dialect_selector = _DIALECT_XPATH_SELECTOR_TEMPLATE.format(
                 dialects_text=" or ".join(
-                    f'contains(text(), "{d.strip()}")' for d in dialect.split("|")
+                    f'contains(text(), "{d.strip()}")'
+                    for d in dialect.split("|")
                 )
             )
             logging.info("Dialect(s): %r", dialect)

--- a/tests/test_wikipron/test_config.py
+++ b/tests/test_wikipron/test_config.py
@@ -153,8 +153,8 @@ def test_ipa_regex(narrow, ipa_regex, word_in_ipa):
                 "  and\n"
                 '  span[@class = "IPA"]\n'
                 "  and\n"
-                '  (span[contains(@class, "ib-content")]//a[text() = "US"]\n'  # noqa: E501
-                '   or span[contains(@class, "ib-content") and (text() = "US")]\n'  # noqa: E501
+                '  (span[contains(@class, "ib-content")]//a[contains(text(), "US")]\n'  # noqa: E501
+                '   or span[contains(@class, "ib-content") and (contains(text(), "US"))]\n'  # noqa: E501
                 '   or count(span[contains(@class, "ib-content")]) = 0)\n'  # noqa: E501
                 "]\n"
             ),
@@ -171,8 +171,8 @@ def test_ipa_regex(narrow, ipa_regex, word_in_ipa):
                 "  and\n"
                 '  span[@class = "IPA"]\n'
                 "  and\n"
-                '  (span[contains(@class, "ib-content")]//a[text() = "General American" or text() = "US"]\n'  # noqa: E501
-                '   or span[contains(@class, "ib-content") and (text() = "General American" or text() = "US")]\n'  # noqa: E501
+                '  (span[contains(@class, "ib-content")]//a[contains(text(), "General American") or contains(text(), "US")]\n'  # noqa: E501
+                '   or span[contains(@class, "ib-content") and (contains(text(), "General American") or contains(text(), "US"))]\n'  # noqa: E501
                 '   or count(span[contains(@class, "ib-content")]) = 0)\n'  # noqa: E501
                 "]\n"
             ),

